File: Intermediate/algorithms/linear_search.py
def linear_search(array,key):
    for i in range(len(array)):
        if array[i]==key:
            return i
    return -1
n=int(input('Enter Size of Array: '))

arr=[input('Enter Element %d : '%(i+1)) for i in range(n)]
key=input('Enter Key : ')

ans=linear_search(arr,key)
if ans==-1:
    print ('ELEMENT NOT FOUND')
else:
    print ('ELEMENT FOUND AT POSITION %d'%(ans+1))

File: Intermediate/algorithms/linear_search.cpp
#include <iostream>

int linear_search(int ar[], int n, int num) {
  int i;
  for(i=0;i<n;i++) {
    if(ar[i] == num) {
      return i;
    }
  }
  return -1;
}

int main(){
  int n;
  std::cout << "Enter number of elements:";
  std::cin >> n;
  int ar[n];
  
  std::cout << "Enter elements:";
  
  for(int i=0;i<n;i++) {
    std::cin >> ar[i];
  }
  
  int num;
  std::cout << "Enter number to look for:";
  std::cin >> num;
  
  int index = linear_search(ar, n, num);
  
  if(index == -1) {
    std::cout << num << " not found";
  }
  else {
    std::cout << num << " found on position " << index+1;
  }
}
